### PR TITLE
Vignette on using QGIS expressions with qgisprocess

### PR DESCRIPTION
**Pull request by florisvdh**
This vignette more or less depends on QGIS>=3.30 for some more involved cases. Some new features were recently added to QGIS to make more things possible in this area from R (https://github.com/qgis/QGIS/issues/50481 & PR https://github.com/qgis/QGIS/pull/51293; https://github.com/qgis/QGIS/issues/50482 & PR https://github.com/qgis/QGIS/pull/51412).

@jannes-m if you can propose other examples, feel free to do so.

Task list:

- [x] add example(s) for data-defined overrides

### CODE REVIEW

File: man/vignette_childs/_qgis_expressions.Rmd [hunk 1]
```diff
@@ -0,0 +1,145 @@
...
+qgis_arguments("native:fieldcalculator") |> subset(name == "FORMULA")
+```
+
+Secondly, one can use expressions for _data-defined_ overriding.
+This means, that an algorithm argument that is usually a fixed (numeric or boolean) value can _also_ take on the value of another field or the result of an expression.
+In the QGIS processing dialog, such arguments have a '[data-defined override](https://docs.qgis.org/latest/en/docs/user_manual/introduction/general_tools.html#data-defined)' button.
+An example is provided by the `DISTANCE` argument of `native:buffer`, for which we query the acceptable values below.
+
+```{r}
+qgis_arguments("native:buffer") |>
```
Comment (jannes-m) [inline man/vignette_childs/_qgis_expressions.Rmd:17]: With QGIS 3.28.2 this results it:

```r
|acceptable_values |
|:-----------------|
|A numeric value   |
```
Not sure, if this is the desired result

Comment (jannes-m) [reply man/vignette_childs/_qgis_expressions.Rmd:17]: ah, later on you say that this is only possible with QGIS 3.30 onwards, ok

**Review by jannes-m — APPROVED** [review]
Thanks for sharing this really interesting and useful vignette!

**Comment by florisvdh** [comment]
Thanks for reviewing and updating!

**Pull request merged**